File: src/sumLeaf.py
class sumLeaf:
    def sumLeaf(self, root:TreeNode) -> int:
        paths = []

        if not root:
            return 0

        def dfs(path, node, paths):
            if not root:
                return
            elif not node.left and not node.right:
                paths.append(path)
            else:
                if node.left:
                    dfs(path + [node.left.val], node.left, paths)
                if node.right:
                    dfs(path + [node.right.val], node.right, paths)

        dfs([root.val], root, paths)
        num = 0

        for path in paths:
            str_path = ''.join([str(x) for x in path])
            num += int(str_path, 2)

        return num

File: src/sumLeaf.java
public class sumLeaf {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    int sum = 0;

    public int sumRootToLeaf(TreeNode root) {
        calSum(root, "");
        return sum;
    }

    public void calSum(TreeNode node, String path) {
        if (node == null)
            return;
        if (node.left == null && node.right == null)
            sum += Integer.parseInt(path + node.val, 2);

        calSum(node.left, path + node.val);
        calSum(node.right, path + node.val);
    }
}